chainer/backends/cuda.py

```diff
@@ -108,11 +107,17 @@ class Event(object):  # type: ignore # for type testing
     class Stream(object):  # type: ignore # for type testing
         pass
 
-    # for `xp is cuda.cupy` and `cuda.libcudnn` to always work
+    # for `xp is chainer.backends.cuda.cupy` to always work
     cupy = object()
-    libcudnn = object()
+
 
 if available:
+    try:
+        libcudnn = cupy.cuda.cudnn  # type: tp.Any # NOQA
```

Comment:
How about doing this only if `cudnn_enabled is True`?
(In that case I think this assignment should succeed?)

Reply:
Ok, it could raise an error even if `cudnn = cupy.cudnn` succeeds
(because `cupy.cuda.cudnn` link to external libraries).